### ISSUE TITLE
Duplicate last name when there's multiple authors

### ISSUE THREAD

**Issue by nawroth** [asciidoctor/asciidoctor.js#159]
With the version in master (and some others post 1.5.2) the last name of authors gets duplicated if there's more than one author.

On a web page where Asciidoctor is loaded, I'm doing:

    Opal.Asciidoctor.$load('= Authors\nNumber One; Number Two\n').attributes.smap

And among the attributes I'm finding:

```
author: "Number One One"
author_1: "Number One One"
author_2: "Number Two Two"
authorcount: 2
authorinitials: "NOO"
authorinitials_1: "NOO"
authorinitials_2: "NTT"
authors: "Number One One, Number Two Two"
```

Using 1.5.2 and doing:

    Opal.Asciidoctor.$load('= Authors\nNumber One; Number Two\n').attributes.map

I'm getting:

```
author: "Number One "
author_1: "Number One "
author_2: "Number Two "
authorcount: 2
authorinitials: "NO"
authorinitials_1: "NO"
authorinitials_2: "NT"
authors: "Number One , Number Two "
```

Ghost spaces now replaced by duplicate last name?!

_I wasn't able to reproduce using asciidoctor.rb 1.5.3, that's why I'm filing the issue here._
(I looked at the HTML output in that case, which was correct.)

**Comment by mojavelinux** [comment]
This is very likely another change caused by "fixes" to regular expressions in Opal. For awhile, Opal was handling certain back-references incorrectly, so we accommodated that quirk. When they fixed it, we started getting the reverse problem. It's been hard to know all the affected regions. This appears to be another.

**Comment by mojavelinux** [comment]
Actually, it looks something different, a spec bug in Opal. More info coming.

**Comment by mojavelinux** [comment]
Here's the problem, demonstrated with an example:

```
opal> 'one two'.split ' ', 3
=> ["one", "two", "two"]
```

The third entry should be nil, but instead it is repeating the previous entry. (Even stranger behavior occurs if we increase 3 to a higher number).

Here's what it should be:

```
irb> 'one two'.split ' ', 3
=> ["one", "two"]
```

**Comment by nawroth** [comment]
@mojavelinux Great that you already tracked down the problem!

**Comment by mojavelinux** [comment]
We should definitely see that this is fixed in Opal, but I'm still going to change this line for 1.5.4 since I found a better way to write it anyway.

**Comment by mojavelinux** [comment]
:muscle:

**Comment by mojavelinux** [comment]
Reported upstream as https://github.com/opal/opal/issues/1267. Fix applied to core as https://github.com/asciidoctor/asciidoctor/commit/d1f451727adc8f04b781c91d59c6ae220fd29e30.

**Comment by Mogztter** [comment]
Thanks @mojavelinux for your analysis. If someone want to contribute a test case to avoid regression, that would be fantastic :smile:

**Comment by Mogztter** [comment]
Resolved in 7710ee25934e7abce0201e26e9982aa2a3e0b4f1
